lua/spectre_common.lua

```diff
@@ -93,8 +112,13 @@ local function determine_if_cacheable(url, namespace, request_headers)
         return cacheability_info
     end
 
+    local http_method = ngx.req.get_method()
     for cache_name, cache_entry in pairs(spectre_config['cached_endpoints']) do
-        if ngx.re.match(url, cache_entry['pattern']) then
+        if ngx.re.match(url, cache_entry['pattern']) and (
+            -- Compute if http_method is same as in config or http method is GET
+            cache_entry['request_method'] == http_method or DEFAULT_REQUEST_METHOD == http_method
```

Comment:
Wouldn't it be easier to set `cache_entry['request_method`'] to the configured value (if present) or DEFAULT_REQUEST_METHOD? That way you don't have to worry about this here and your condition becomes: 

```lua
if ngx.re.match(url, cache_entry['pattern']) and http_method == cache_entry['request_method'] then
    ...
end
```
(don't you already do this line 101 above?)